services/wfm/src/main/java/org/openkilda/wfm/topology/isllatency/bolts/IslLatencyBolt.java

```diff
@@ -0,0 +1,170 @@
+/* Copyright 2019 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.isllatency.bolts;
+
+import org.openkilda.messaging.Message;
+import org.openkilda.messaging.Utils;
+import org.openkilda.messaging.info.InfoData;
+import org.openkilda.messaging.info.InfoMessage;
+import org.openkilda.messaging.info.event.IslOneWayLatency;
+import org.openkilda.messaging.info.event.IslRoundTripLatency;
+import org.openkilda.model.Isl;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.error.IllegalIslStateException;
+import org.openkilda.wfm.error.IslNotFoundException;
+import org.openkilda.wfm.error.JsonDecodeException;
+import org.openkilda.wfm.error.SwitchNotFoundException;
+import org.openkilda.wfm.topology.AbstractTopology;
+import org.openkilda.wfm.topology.isllatency.LatencyAction;
+import org.openkilda.wfm.topology.isllatency.service.DecisionMakerService;
+import org.openkilda.wfm.topology.isllatency.service.IslLatencyService;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Tuple;
+
+import java.io.IOException;
+
+@Slf4j
+public class IslLatencyBolt extends AbstractBolt {
+    private final PersistenceManager persistenceManager;
+    private transient IslLatencyService islLatencyService;
+    private transient DecisionMakerService decisionMakerService;
+
+    public IslLatencyBolt(PersistenceManager persistenceManager) {
+        this.persistenceManager = persistenceManager;
+    }
+
+    @Override
+    protected void init() {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        islLatencyService = new IslLatencyService(transactionManager, repositoryFactory);
+        decisionMakerService = new DecisionMakerService(repositoryFactory);
+    }
+
+    private String getJson(Tuple tuple) {
+        return tuple.getString(0);
```

Comment:
Wouldn't it be better to use the 'pullValue ()' method?

Reply:
fixed